.github/workflows/upgrade-tests.yaml

```diff
@@ -0,0 +1,89 @@
+name: Upgrade Tests
+on: [push, pull_request, workflow_dispatch]
+
+jobs:
+  upgrade-tests:
+    runs-on: ubuntu-latest
+    timeout-minutes: 90
+    concurrency:
+      group: ${{ github.workflow }}-${{ github.ref }}
+      cancel-in-progress: true
+
+    env:
+      BRANCH_NAME: ${{ github.event.inputs.branch || 'master' }}
+
+    steps:
+      - name: Checkout noobaa-core (master/input branch)
+        uses: actions/checkout@v4
+        with:
+          repository: 'noobaa/noobaa-core'
+          path: 'noobaa-core'
+          ref: ${{ env.BRANCH_NAME }}
+
+      - name: Deploy minikube
+        run: |
+          cd ./noobaa-core
+          sudo bash ./.travis/deploy_minikube.sh
+
+      - name: Build noobaa
+        run: |
+          cd ./noobaa-core
+          make noobaa NOOBAA_TAG=noobaa-core:${{ env.BRANCH_NAME || 'master' }}
+
+      - name: Checkout noobaa-core (latest)
+        uses: actions/checkout@v4
+        with:
+          repository: 'noobaa/noobaa-core'
+          path: 'noobaa-core'
+
+      - name: Build noobaa (latest)
+        run: |
+          cd ./noobaa-core
+          make noobaa NOOBAA_TAG=noobaa-core:upgrade-tests
```

Comment:
If we are on master branch, then the image of `noobaa-core:master` and `noobaa-core:upgrade-tests` are the same? (maybe I missed something - why should we continue with the job?)

Reply:
Here, the flow is like in step 1 we are building noobaa core from master/input branch for installation of noobaa and in step 4 we are building noobaa core using PR code to upgrade the noobaa.